colcon_core/shell/template/prefix_util.py.em

```diff
@@ -295,10 +295,10 @@ env_state = {}
 def _prepend_unique_value(name, value):
     global env_state
     if name not in env_state:
-        if name not in os.environ:
-            env_state[name] = set()
-        if os.environ.get(name):
+        if os.environ.get(name, ''):
```

Comment:
The default `default` value of `get()` is `None` and is sufficient for this check. So no need to change this line.

Reply:
That's true, but I think this formulation makes it clearer that missing environment variables are treated as if they contained the empty string. However, this is a minor stylistic thing, so if you find it irritating I can remove the default value.

Reply:
If I read the `, ''` I am asking myself why is it there? Why is the default value `None` not sufficient. Since there is no need for it I rather don't want to think about it :wink:

Thanks for updating it.